### ISSUE TITLE
Make Upload Progress Alerts more friendly to Screen Readers (blocked by 8156)

### ISSUE THREAD

**Issue by JessicaMarine** [ustaxcourt/ef-cms#7812]
Currently when uploading documents, screen reader will repeat upload message multiple times instead of announcing actual progress. 

This could be refined using `progressbar` Aria roles to make screen reading for uploads more insightful, and less verbose and redundant. [Learn more](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role#:~:text=The%20progressbar%20role%20should%20be,toward%20completing%20the%20requested%20action.).

#### Recommendation
Recommended screen reading would be announcing a custom alert message once ("Less Than 1 Minute Left"), then announcing actual percent progress as number changes. 

#### Steps to Reproduce
- turn on VoiceOver for Mac
- complete creating a case
- upload a case, listen to progress alerts from modal

**Comment by kkoskelin** [comment]
I'm going to have to put this one on hold pending feedback from 18F.
* I am unsure what "announcing actual progress" is supposed to sound like. I've attempted to configure VO for MacOS to do so without success.  I am not a daily user of VO nor do I depend on it and have limited experience.
* any documentation that could be provided would be helpful to 1) configure VO to hear progress, and 2) a demo page where this can be experienced
* I have tried to create a screenreader-only progress bar component but also cannot get it to announce progress in any manner. I have likewise searched for advice on the above but have only seen examples which behave similar to the current progress bars.

**Comment by mariahkannenberg** [comment]
Blocked - waiting for response from 18F

**Comment by adunkman** [comment]
An update here — attempting to reproduce and it wasn’t immediately reproducible. Reaching out to a VoiceOver expert for confirmation and will update here with the response.

**Comment by adunkman** [comment]
Here’s a video of the problem — I reproduced by starting a new case signed in as a petitioner (be sure to unmute the video).

https://user-images.githubusercontent.com/14930/114769858-bc6e3280-9d38-11eb-8ed3-72749eab2396.mov

There are a few things contributing to this experience: 

- The `aria-live` element is the entire modal, which means when the modal is re-rendered, the screen reader attempts to read the entire dialog, including a close button (which may not be appropriate — it doesn’t look like this is closable without canceling the upload?). Using an element which is just the progress bar or status text as `aria-live` will read only the status text when it changes. 
- Defining the progress bar’s endpoints (either using `<progress max="100" min="0" value="24">` or `<div aria-role="progressbar" aria-valuemax="100" aria-valuemin="0" aria-valuenow="24">`) and focusing that element will cause the screenreader to indicate progress non-verbally (in VoiceOver’s case, ticking).
- Throttling how often the status updates are written to the DOM allows the screenreader to complete its sentences.

Here’s a [JSFiddle](https://jsfiddle.net/gna93p6t/7/) with some code demonstrating both approaches (using a `<progress>` element and a `<div>`) and the video of how it sounds (again, don’t forget to unmute):


https://user-images.githubusercontent.com/14930/114908470-21ce2c00-9dea-11eb-864b-855bc9c36763.mov

**Comment by adunkman** [comment]
@kkoskelin does this get you unstuck? Happy to chat further or pair!

**Comment by kkoskelin** [comment]
Excellent, thank you.

**Comment by adunkman** [comment]
Reviewed in `test` just now — the progress alerts are much better than before — the alert is announced only when the time estimate changes (be sure to unmute):

https://user-images.githubusercontent.com/14930/115452094-ff724f00-a1eb-11eb-918f-daefbd3b3d76.mov

However, this means that for blind users who cannot see the progress bar, there is no indication of work being performed for a minute at a time. If we focus the progress element in addition to these changes, there is an audible tick when the progress bar changes: 


https://user-images.githubusercontent.com/14930/115452652-afe05300-a1ec-11eb-8e4a-23e007af7491.mov

**Comment by adunkman** [comment]
I still don’t see focus assigned to the `<progress>` element — which means there aren’t ticks notifying the user of the percentage progress. Maybe there hasn’t been a deploy to `test` since the last … test? 

https://user-images.githubusercontent.com/14930/115933690-5d599d80-a45d-11eb-96cc-a5a3ab4765c5.mov

**Comment by adunkman** [comment]
I’m not able to verify the progress element is focused at https://app.test.ef-cms.ustaxcourt.gov/file-a-petition/step-5 — @kkoskelin are you seeing the progress element focused there? 

Give me a shout if it’s easier to hop on a call. My understanding was that with your latest changes, I should be seeing the progress element focused.

**Comment by mariahkannenberg** [comment]
Check w/Mike - merged to test yet? @mmarcotte

**Comment by adunkman** [comment]
Tested again just now and still not seeing focus applied to the `<progress>` element. As shown in the video, the `test` environment was last deployed `Tue, Apr 27, 2021, 10:31 AM EST`.

https://user-images.githubusercontent.com/14930/116306247-d2d1b080-a772-11eb-8df3-56abaeceb760.mov

**Comment by adunkman** [comment]
Tested and verified fixed! 🎉 

Thanks for all your work here, @kkoskelin! 

https://user-images.githubusercontent.com/14930/116568622-9e770500-a8d6-11eb-91a2-cef3f9b7b9ab.mov

**Comment by JessicaMarine** [comment]
Blocked by 8156 because its branch has some commits merged into it. Have to wait until BOTH are ready for staging deploy
